lrpar/tests/build.rs

```diff
@@ -0,0 +1,57 @@
+use cfgrammar::yacc::{YaccKind, YaccOriginalActionKind};
+use glob::glob;
+use lrlex::LexerBuilder;
+use lrpar::CTParserBuilder;
+use std::{env, fs, io::Read, path::PathBuf};
+use yaml_rust::YamlLoader;
+
+fn main() -> Result<(), Box<std::error::Error>> {
+    let out_dir = env::var("OUT_DIR").unwrap();
+    for entry in glob("src/*.test")? {
+        match entry {
+            Ok(path) => {
+                if path.is_file() {
+                    // Parse test file
+                    let mut file = fs::File::open(&path)?;
+                    let mut s = String::new();
+                    file.read_to_string(&mut s)?;
+                    let docs = YamlLoader::load_from_str(&s).unwrap();
+                    let grm = &docs[0]["grammar"].as_str().unwrap();
+                    let lex = &docs[0]["lexer"].as_str().unwrap();
+                    let yacckind = match docs[0]["yacckind"].as_str().unwrap() {
+                        "NoAction" => YaccKind::Original(YaccOriginalActionKind::NoAction),
+                        "UserAction" => YaccKind::Original(YaccOriginalActionKind::UserAction),
+                        "Grmtools" => YaccKind::Grmtools,
+                        "GenericParseTree" => {
+                            YaccKind::Original(YaccOriginalActionKind::GenericParseTree)
+                        }
+                        s => panic!("Yacckind '{}' not supported", s)
+                    };
+
+                    // Create grammar files
+                    let base = path.file_stem().unwrap().to_str().unwrap();
+                    let mut pg = PathBuf::from(&out_dir);
+                    pg.push(format!("{}.y", base));
+                    fs::write(&pg, &grm).unwrap();
+                    let mut pl = PathBuf::from(&out_dir);
+                    pl.push(format!("{}.l", base));
+                    fs::write(&pl, &lex).unwrap();
+
+                    // Build parser and lexer
+                    let lex_rule_ids_map = CTParserBuilder::new()
+                        .yacckind(yacckind)
+                        .process_file(pg.to_str().unwrap(), &out_dir)?;
```

Comment:
Deosn't `process_file` take a file (not a directory) as its second arg?

Reply:
Only `LexerBuilder` apparently. `CTParserBuilder` takes a directory and generates the `_y.rs` filename automatically, it appears.

Reply:
Hmm, we should make those two consistent (one way or another -- I don't feel strongly). Can you maybe make a PR after this one to do that?

Reply:
Will do.